scandir_windows.go

```diff
@@ -83,17 +96,20 @@ func (s *Scanner) done(err error) {
 		return
 	}
 
-	if cerr := s.dh.Close(); err == nil {
-		s.err = cerr
+	s.err = err
```

Comment:
Here named return parameter `err` seems to be always `nil`

So the following `s.err == nil` at `L101` is always true

On 32-bit Windows, `err = s.dh.Close()` sometimes returns EOF inexplicably, 
which is then assigned to `s.err` at `L102`

This seems to be the cause for https://github.com/influxdata/telegraf/issues/11823
and https://github.com/karrick/godirwalk/issues/70

@karrick PTAL

@powersj You may also be interested?